server/api/auth/google.js

```diff
@@ -0,0 +1,87 @@
+const passport = require('passport');
+const GoogleStrategy = require('passport-google-oauth').OAuth2Strategy;
+const loginWithIdp = require('./loginWithIdp');
+
+const radix = 10;
+const PORT = parseInt(process.env.REACT_APP_DEV_API_SERVER_PORT, radix);
+const rootUrl = process.env.REACT_APP_CANONICAL_ROOT_URL;
+const clientID = process.env.REACT_APP_GOOGLE_CLIENT_ID;
+const clientSecret = process.env.GOOGLE_CLIENT_SECRET;
```

Comment:
Wondering what this is used for. The Google Web login reference does not mention using the client secret: https://developers.google.com/identity/sign-in/web/reference. Apparently this is [used to fetch user profile info](https://github.com/jaredhanson/passport-oauth2/blob/fda4a53fe63ac13b825aec0ae87a4d6bb599ebd5/lib/strategy.js#L95) but shouldn't the access token suffice for that.

What if this is left out? Does fetching the profile fail?

Reply:
I tested this without clientSecret and it didn't work. At least the Passport.js expects that the client secret is there. After some investigation I think I found the answer from [Google docs](https://developers.google.com/identity/protocols/oauth2#installed): For example, a JavaScript application does not require a secret, but a web server application does.

It's a little confusing because we are also using JavaScript in the FTWs server. But anyway, when we don't use Google's JavaScript client directly for the web application but we are handling the authentication in the web server, we need to provide the client secret too.